app/src/main/java/com/example/playlistmaker/search/ui/SearchActivity.kt

```diff
@@ -69,6 +67,7 @@ class SearchActivity : AppCompatActivity(), TrackViewHolder.OnItemClickListener
         }
 
         binding.updateButton.setOnClickListener {
+            val lastInput = (viewModel.observeState().value as SearchActivityState.NoConnection).lastInput
```

Comment:
Лучше делать safe cast:
```suggestion
            val lastInput = (viewModel.observeState().value as? SearchActivityState.NoConnection)?.lastInput ?: return@setOnClickListener
```